### ISSUE TITLE
Show used modules in scan report

### ISSUE THREAD

**Issue by haerter-tss** [mercedes-benz/sechub#2546]
### Situation
Currently the reports do not indicate which modules (codeScan, webScan, secretScan, etc) have been used, when the findings are empty. This can lead to uncertainty for the user. 

### Example
A project uses the codeScan module. After a while the administrator also adds the secretScan module to that project. The user has no way of knowing that their code now runs through a second module.

### Solution
The sechub report should indicate which modules were used.

**Comment by zigfridus** [comment]
@haerter-tss @Jeeppler 
I would like to precise. Does this issue was resolved with help of #345 ?

**Comment by Jeeppler** [comment]
@zigfridus yes, I think you are right. However, the problem is, that if no issues were found you do not know exactly what modules were active in the scan.

**Comment by zigfridus** [comment]
@Jeeppler 
Do you mean that in the attched picture should be some information about used scan modules?
![Screenshot 2024-03-20 150135](https://github.com/mercedes-benz/sechub/assets/4904593/2cf656f5-a118-4bc4-ade4-f0df9ac4c10a)

**Comment by de-jcup** [comment]
The report shall have the information which scan types where used for the report.

To handle this we will:
1. Track the scan types inside the runtime execution context (SecHub scan domain)
2. The final result will be set into the SERECO product result
3. The SecHub report model will be extended at meta level with runtime information about used scan types => available inside JSON report
4. Same for open api pendant for model info
5. HTML report will be extended (scan type info will appear before user messages)
6. Web UI report page will be extended (scan type info will appear before user messages)

**Comment by de-jcup** [comment]
JSON output will be as following:
```json
{
  "metaData" : {
    "labels" : { },
    "summary" : {
      "codeScan" : {
        "total" : 109,
        "critical" : 0,
        "high" : 0,
        "medium" : 1,
        "low" : 105,
        "unclassified" : 0,
        "info" : 3,
        "details" : {
          "medium" : [ {
            "cweId" : 36,
            "name" : "Absolute Path Traversal",
            "count" : 1
          } ],
          
          "info" : [ {
            "cweId" : 778,
            "name" : "Insufficient Logging of Exceptions",
            "count" : 3
          } ]
        }
      }
    },
    "executed" : [ "codeScan" ]
// ...
}
```
